### ISSUE TITLE
bug(types): Typing files are missing for `tseslint.configs.strictTypeChecked` in FlatConfig

### ISSUE THREAD

**Issue by motss** [azat-io/eslint-plugin-perfectionist#139]
### Describe the bug

I'm trying to use ESLint's new `FlatConfig` and `typescript-eslint`'s `tseslint.configs.strictTypeChecked` enabled for stricter type checking and linting. However, ESLint reports the lint error below for `eslint-plugin-perfectionist` due to missing types from the package:

```sh
  20:3   error  Unsafe argument of type `any` assigned to a parameter of type `ConfigWithExtends`  @typescript-eslint/no-unsafe-argument
```

I was quite surprised that a project that is entirely written in TypeScript chose to omit type files completely. It'd be great if type files could be added so that ESLint read the typings of the package correctly without the need to add additional rule to turn off some rules in `eslint.config.*js` for all the users.

### Code example

1. Go to https://stackblitz.com/edit/vitest-dev-vitest-c5wksh?file=eslint.config.mjs
2. Wait for dependencies to be installed and you will see linting errors
3. If Step 2 fails, you can manually run `pnpm install && pnpm lint`

### ESLint version

9.1.1

### ESLint Plugin Perfectionist version

2.10.0

### Additional comments

{
    "@eslint/eslintrc": "^3.0.2",
    "@eslint/js": "^9.1.1",
    "eslint": "^9.1.1",
    "eslint-plugin-perfectionist": "^2.10.0",
    "typescript-eslint": "^7.7.1"
}

### Validations

- [X] Read the [docs](https://eslint-plugin-perfectionist.azat.io).
- [X] Check that there isn't [already an issue](https://github.com/azat-io/eslint-plugin-perfectionist/issues) that reports the same bug to avoid creating a duplicate.

**Comment by azat-io** [comment]
Thank you for your question! We plan to add type support in the next major version.

**Comment by azat-io** [comment]
Thank you for your issue!

The types were added in 2b63ab558f431a33fbdfb103c3279ec34011d135 and released in [v3.0.0](https://github.com/azat-io/eslint-plugin-perfectionist/releases/tag/v3.0.0).

If you'd like, you can support the release with a retweet:
https://x.com/azat_io_en/status/1815367279191761054
